sdk/src/frame_impl.cpp

```diff
@@ -101,6 +102,11 @@ aditof::Status FrameImpl::getData(aditof::FrameDataType dataType,
                                   uint16_t **dataPtr) {
     using namespace aditof;
 
+    if (dataPtr == nullptr) {
+        LOG(WARNING) << "Invalid frame dataPtr";
```

Comment:
- this should be an `ERROR`
- In this case, after validating `dataPtr` it would make sense to also validate `*dataPtr`